### PR TITLE
Adjust E2E test codes

### PR DESCRIPTION
**Pull request by qiuming-best**
Thank you for contributing to Velero!

# Please add a summary of your change
- Using random number instead of UUID for some cases failed with long UUID string
- Normalize the codes in E2E test, including names in each test case, respecting the test frame. removing StartRun function
- Deleting some undelete or useless resources, including pull image secrets, and delete resource after each case
- Adding timestamps for analysis time-consuming in some procedures
# Does your change fix a particular issue?

Fixes #(issue)
https://github.com/vmware-tanzu/velero/issues/6098
# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: test/e2e/basic/nodeport.go [hunk 1]
```diff
@@ -66,37 +67,40 @@ func (n *NodePort) CreateResources() error {
...
-	})
+	for _, ns := range *n.NSIncluded {
+		By(fmt.Sprintf("Start to destroy namespace %s......", n.CaseBaseName), func() {
+			Expect(CleanupNamespacesWithPoll(ctx, n.Client, NodeportBaseName)).To(Succeed(),
+				fmt.Sprintf("Failed to delete namespace %s", n.CaseBaseName))
+			Expect(WaitForServiceDelete(n.Client, ns, n.serviceName, false)).To(Succeed(), "fail to delete service")
+			_, err := GetAllService(ctx)
+			Expect(err).To(Succeed(), "fail to get service")
+		})
+		namespaceToCollision := ns + "-tmp"
```
Comment (danfengliu) [inline test/e2e/basic/nodeport.go:96]: I like to have variable namespaceToCollision as a case scope one.

File: test/e2e/basic/nodeport.go [hunk 2]
```diff
@@ -134,37 +138,38 @@ func (n *NodePort) Restore() error {
...
-
-	args = n.RestoreArgs
-	args = append(args[:index], append([]string{restoreName2}, args[index:]...)...)
-	args = append(args, "--preserve-nodeports=true")
-	By(fmt.Sprintf("Start to restore %s with nodeports preservation ......", restoreName2), func() {
-		Expect(VeleroRestoreExec(ctx, n.VeleroCfg.VeleroCLI, n.VeleroCfg.VeleroNamespace,
-			restoreName2, args, velerov1api.RestorePhaseCompleted)).To(Succeed(), func() string {
-			RunDebug(context.Background(), n.VeleroCfg.VeleroCLI, n.VeleroCfg.VeleroNamespace, "", restoreName2)
-			return "Fail to restore workload"
+	for _, ns := range *n.NSIncluded {
```
Comment (danfengliu) [inline test/e2e/basic/nodeport.go:141]: `NSIncluded` seems useless for this case, that's why I add a `namespace` case scope variable in this case, I think the basic type should cover this mode as no multi-namespace is involved according  test intention.

File: test/e2e/basic/nodeport.go (hunk 1 above)
Comment (qiuming-best) [reply test/e2e/basic/nodeport.go:96]: I've added it back into NodePort

File: test/e2e/basic/nodeport.go (hunk 2 above)
Comment (qiuming-best) [reply test/e2e/basic/nodeport.go:141]: Our current  E2E are all kinds of variable names.

For the test, we want only input CaseBaseName and NamespacesTotal, all other variables will have standard assigned names. which will align all names in each case.

for the namespace is better just generated by NamespacesTotal for alignment

**Review by danfengliu — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**